subsys/bluetooth/mesh/access.c

```diff
@@ -116,9 +116,14 @@ static int32_t next_period(struct bt_mesh_model *mod)
 	struct bt_mesh_model_pub *pub = mod->pub;
 	uint32_t elapsed, period;
 
-	period = bt_mesh_model_pub_period_get(mod);
-	if (!period) {
-		return 0;
+	if (mod->pub->count) {
+		period = BT_MESH_PUB_TRANSMIT_INT(mod->pub->retransmit);
+		period *= BT_MESH_PUB_MSG_NUM(mod->pub);
+	} else {
+		period = bt_mesh_model_pub_period_get(mod);
+		if (!period) {
+			return 0;
+		}
```

Comment:
Should perform this check in the first branch too, just to make sure the handling of changing retransmit counts is coherent. In that branch, we should probably fall back to this branch though, so we can schedule the next period if the count has changed.